### DISCUSSION TITLE
Can't pass the error Minimum OS Version required when uploading my IPA

### DISCUSSION THREAD

**Discussion by stephanberger4** [airsdk/Adobe-Runtime-Support#1497] [Q&A] [answered]
Hi, 

I'm trying to upload my AIR app to the app store. I've got this error : 
![error](https://user-images.githubusercontent.com/6382334/147403098-3bb5547c-5870-4345-a5e8-c8b4aec0a166.png)

So, basically, the first error is saying that the xml contains a 'MinimumOSVersion' key that is not supported on IOS and the second error is asking me to specified the minimum OS version..... 

Here's what I've put in my xml before producing IPA file :  

```
 <Entitlements><![CDATA[
<key>application-identifier</key>
<string>L8N8KB56L6.nc.cocogeek.mareesNC</string>
<key>MinimumOSVersion</key>
<string>10.0</string>
<key>aps-environment</key>
<string>production</string>
...
```

And here's the Info.plist when I open with WinRar my IPA file : 

```
.....
<key>DTXcode</key>
        <string>1300</string>
        <key>DTXcodeBuild</key>
        <string>13A233</string>
        <key>LSRequiresIPhoneOS</key>
        <true/>
        <key>MinimumOSVersion</key>
        <string>9.0</string>
        <key>NSMainNibFile</key>
        <string>MainWindow</string>
        <key>NSMainNibFile~ipad</key>
.....
```

So, I don't understand why I've got these two errors....

Not sure why the  Info.plist is changing the version that I have put in my xml file too... (10.0 versus 9.0 in the Info.plist)

Any ideas ? 

Thank you very much

**Comment by ajwfrost** [comment, marked as answer]
Which version of the AIR SDK have you been using for this? There were some changes made in October around the collation of the minimum OS version.

But, I don't think the minimum iOS version should be put into the `Entitlements` block of the application descriptor file, it should be in the `InfoAdditions` block...

**Reply by stephanberger4** [reply]
I've tried to put the minimum iOS version into `InfoAdditions `block and it seems you're right as I don't have the first error. But I still have the second one... 

![image](https://user-images.githubusercontent.com/6382334/147622923-86f59ed8-d6d9-4686-a008-d15fd197592e.png)

**Reply by ajwfrost** [reply]
Interesting, so it seems the info.plist value may not correspond to the value in the executable...
Is it possible to get hold of the IPA file? you could upload to https://transfer.harman.com/requests/dvb4IkbZFVvsj8tlS2pIsL
Plus if you're able to set the ADT configuration file so that it outputs a log file (`AIR_SDK_folder/lib/adt.cfg` and enable `DebugOut=true`) then send us the log there (`~/adt.log`) it would be good to check the linker command..

thanks

**Reply by ajwfrost** [reply]
Thanks for those. Looking at the first one, with the value set to 10 in the app descriptor file, you have an Info.plist which copies this:
```
        <key>MinimumOSVersion</key>
        <string>10.0</string>
```
and the executable has:
```
Load command 9
      cmd LC_VERSION_MIN_IPHONEOS
  cmdsize 16
  version 10.0
      sdk 15.0
```
So that all looks good .. are you getting the bundle error with this build?


With the OS 9 version:
```
        <key>MinimumOSVersion</key>
        <string>9.0</string>
```
is copied into the Info.plist file, but the executable has:
```
Load command 9
      cmd LC_VERSION_MIN_IPHONEOS
  cmdsize 16
  version 10.0
      sdk 15.0
```

So there is a discrepancy between the value in the XML file and Info.plist file, and the value used by the linker (and the log also shows this, ` -ios_version_min 10.0` at the end of the of the `ld64` command-line).

And the reason for this is found in the extensions' platform XML files:
```
META-INF/AIR/extensions/com.myflashlab.air.extensions.dependency.androidx.core/META-INF/ANE/iPhone-ARM/platform.xml:        <option>-ios_version_min 10.0</option>
META-INF/AIR/extensions/com.myflashlab.air.extensions.zipManager/META-INF/ANE/iPhone-ARM/platform.xml:        <option>-ios_version_min 10.0</option>
META-INF/AIR/extensions/com.myflashlab.air.extensions.permissionCheck/META-INF/ANE/iPhone-ARM/platform.xml:        <option>-ios_version_min 10.0</option>
```

So ... because the extensions have a minimum version of 10, this is what's being set up in the binary file - but we're not pulling this value in to the Info.plist file. Which is a bug, and I hope we will be able to parse the extension platform details prior to generating the Info.plist in order to fix this... 

Workaround is to set the minimum IOS level manually as it looks like you have done with the other package (`...811_minOS_10.ipa`)?

thanks

**Reply by stephanberger4** [reply]
You're right ! I've just tried ot upload the build with the 10 version and no errors ! It was uploaded to the App Store !! (I'm still wondering why I didn't try it before sending it to you.... sorry !). 
The other one (version 9) doesn't work. 

Thank you again for your precious help !
